tests/unit/lms/services/d2l_api/client_test.py

```diff
@@ -3,7 +3,73 @@
 import pytest
 
 from lms.services.d2l_api._basic import BasicClient
-from lms.services.d2l_api.client import D2LAPIClient
+from lms.services.d2l_api.client import D2LAPIClient, D2LTableOfContentsSchema
+from tests import factories
+
+
+class TestD2LTableOfContentsSchema:
+    def test_valid(self, table_of_contents_response):
+        schema = D2LTableOfContentsSchema(
+            factories.requests.Response(json_data=table_of_contents_response)
+        )
+
+        result = schema.parse()
+
+        assert result == {
+            "modules": [
+                {
+                    "topics": [
+                        {
+                            "id": "ID 1",
+                            "name": "Title 1",
+                            "updated_at": "Date 1",
+                            "type": "File",
+                        }
+                    ],
+                    "modules": [
+                        {
+                            "topics": [
+                                {
+                                    "id": "ID 1",
+                                    "name": "Title 1",
+                                    "updated_at": "Date 1",
+                                    "type": "File",
+                                }
+                            ],
+                        }
+                    ],
+                }
+            ]
+        }
```

Comment:
:shrug: This is kind of painful, but with something this nested, and with every key different I'm not sure what you can do?